### PR TITLE
feat: improve procedural macro safety, error handling, and DX in elif-http-derive

### PR DESCRIPTION
**Pull request by krcpa**
## Summary

This comprehensive enhancement addresses issue #260 by significantly improving the procedural macros in `elif-http-derive` with better error handling, enhanced validation, and improved developer experience.

#### Enhanced Error Messages & Developer Experience
- ✅ Improved parameter validation error messages with actionable hints
- ✅ Enhanced error context using better `syn::Error::new_spanned` usage  
- ✅ More specific guidance for common mistakes like type mismatches
- ✅ Added `get_recommended_param_type` helper for contextual error suggestions

#### Comprehensive Compile-Time Validation  
- ✅ Added `validate_method_consistency` function with 6 validation checks:
  1. Route path parameters must have corresponding `#[param]` declarations
  2. `#[param]` declarations must correspond to route path parameters
  3. Function parameters must match route and `#[param]` declarations
  4. Prevents redundant `#[request]` attribute usage
  5. Body parameter validation for consistency
  6. Detection of unexpected/unhandled parameters

#### Improved Error Message Examples
**Before:**
```
error: Parameter type mismatch: Parameter 'id' has type 'u32' but expected one of: String, &str, str
```

**After:**
```
error: Parameter 'id' has type 'u32' but expected one of: String, &str, str. 
Hint: Change the function parameter to match #[param(id: string)] or change the #[param] declaration to match the function parameter.
```

### 🔧 Technical Implementation

- **Enhanced validation pipeline**: Route path → `#[param]` declarations → Function signature consistency
- **Improved parameter type inference** and validation with detailed error context
- **Better unsupported parameter detection** with precise error location
- **Comprehensive method validation** covering all macro interaction scenarios

### 📁 Files Changed

- `src/http_methods.rs`: Added `validate_method_consistency` function and enhanced validation pipeline (+132 lines)
- `src/params.rs`: Improved error messages and added helper functions (+38 lines)  
- `tests/ui/fail/*.stderr`: Updated expected error messages to match improvements

### 🧪 Testing & Compatibility

- ✅ All existing tests pass (19+ test suites including UI tests)
- ✅ Enhanced UI test validation with better error message coverage
- ✅ Zero breaking changes to existing API
- ✅ Maintained full backward compatibility
- ✅ Comprehensive trybuild integration testing both pass and fail scenarios

## Test plan

- [x] Run full test suite: `cargo test` ✅ All tests pass
- [x] Run UI tests: `cargo test --test ui_tests` ✅ All compilation tests pass
- [x] Verify error message improvements in fail scenarios ✅ Enhanced error messages validated
- [x] Test backward compatibility with existing controller examples ✅ No breaking changes
- [x] Validate enhanced compile-time validation catches edge cases ✅ Comprehensive validation working

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: crates/elif-http-derive/src/http_methods.rs [hunk 1]
```diff
@@ -407,4 +431,108 @@ fn get_extraction_method(
...
+                    let param_name = ident.to_string();
+                    let param_type_str = quote! { #pat_type.ty }.to_string();
+                    
+                    if param_type_str.contains("ElifRequest") {
+                        has_request_param = true;
+                    } else {
+                        fn_params.insert(param_name, param_type_str);
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (krcpa) [reply crates/elif-http-derive/src/http_methods.rs:467]: @gemini-code-assist check now please.

[the remark this answers is not in the document]
Comment (krcpa) [reply crates/elif-http-derive/src/http_methods.rs]: @gemini-code-assist check now please.
